compose/ui/ui-text/src/desktopMain/kotlin/androidx/compose/ui/text/platform/AwtFontInterop.kt

```diff
@@ -0,0 +1,161 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.text.platform
+
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.text.ExperimentalTextApi
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import java.awt.Font
+
+/**
+ * Obtain a [FontFamily] corresponding to this AWT [Font]. The Compose
+ * runtime is able to match font families available in the system, and
+ * ones embedded in the JetBrains Runtime (see note below). If the [Font]
+ * is a logical font (i.e., one of [Font.DIALOG], [Font.DIALOG_INPUT],
+ * [Font.SERIF], [Font.SANS_SERIF], or [Font.MONOSPACED]) then the actual
+ * physical font will be resolved.
+ *
+ * If this [Font] is not known to Compose — e.g., because it's been created
+ * from a file — the lookup will return a fallback font family. If you need
+ * to create a font family from font files instead, you can do it like
+ * this:
+ * ```kotlin
+ * val myFontFamily = FontFamily(
+ *     File(myRegularFontFile, FontWeight.Normal, FontStyle.Normal),
+ *     File(myItalicFontFile, FontWeight.Normal, FontStyle.Italic),
+ *     // ...
+ * )
+ * ```
+ *
+ * **Note:** this will return a fallback font family, such as
+ * [FontFamily.Default] or [FontFamily.Monospace], when running on Java
+ * VMs other than the JetBrains Runtime, and/or if the current module
+ * cannot access the `sun.font` module. To ensure the `sun.font` module
+ * is accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ *
+ * @see resolveFontFamilyNameOrNull
+ * @see fontFamilyNameOrNull
+ */
+@ExperimentalTextApi
+fun Font.asFontFamily(): FontFamily {
+    val familyName = resolveFontFamilyNameOrNull() ?: family
+    return FontFamily(familyName)
+}
+
+/**
+ * Get the _preferred font family name_, which should be used instead of
+ * the [Font.getFamily] and `Font2D.familyName` when dealing with Compose.
+ *
+ * **Why is this better than the AWT API?** On Windows, and potentially in
+ * other cases, the [Font.getFamily] returned by AWT contains the style and
+ * weight of the [Font], in addition to the _actual_ font family name. This
+ * can cause issues when trying to look up the corresponding Compose font
+ * families. Besides that, it will result in duplicate entries when listing
+ * font families.
+ *
+ * **Note:** this will return `null` when running on Java VMs other
+ * than the JetBrains Runtime, and/or if the current module cannot
+ * access the `sun.font` module. To ensure the `sun.font` module is
+ * accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ */
+@ExperimentalTextApi
+fun Font.fontFamilyNameOrNull(): String? = AwtFontUtils.getPreferredFontFamilyName(this)
```

Comment:
This, `fontFamilyNameOrNull`, `resolveFontFamilyNameOrNull` seems aren't Compose related. Can we make them private/internal?